src/components/v2/Modal/Modal.tsx

```diff
@@ -0,0 +1,71 @@
+import React, { ReactElement } from 'react';
+import Backdrop from '@mui/material/Backdrop';
+import { Button, Modal as MUIModal, ModalProps } from '@mui/material';
+import Fade from '@mui/material/Fade';
+import Box from '@mui/material/Box';
+import { useModalStyles } from './ModalStyles';
+
+export interface IModalProps extends ModalProps {
+  children: any;
+  className?: string;
+  isOpened: boolean;
+  handleClose: () => void;
+  titleComponent?: ReactElement;
+}
+
+export const Modal: React.FC<IModalProps> = ({
+  children,
+  handleClose,
+  isOpened,
+  titleComponent,
+}) => {
+  const s = useModalStyles({ hasTitleComponent: Boolean(titleComponent) });
+  const closeModal = () => {
+    handleClose();
+  };
+  return (
+    <MUIModal
+      open={isOpened}
+      onClose={closeModal}
+      onBackdropClick={closeModal}
+      closeAfterTransition
+      BackdropComponent={Backdrop}
+      BackdropProps={{
+        timeout: 500,
+      }}
+    >
+      <Fade in={isOpened}>
+        <Box sx={s.box}>
```

Comment:
I'd pass `className` to this tag, so that its width could be modified through CSS for example.